### ISSUE TITLE
Building a NodeOS derivative

### ISSUE THREAD

**Issue by OrdinaryJellyfish** [NodeOS/NodeOS#411]
How would I build a NodeOS derivative? Does NodeOS have any such APIs for this? I would like to build an OS with a desktop GUI (most likely using something like Electron), and NodeOS seems perfect for this, but I can't seem to find any APIs for it that I could use.

**Comment by piranna** [comment]
There's no API like that, but instead several intechangeable layers, so you can modify and reuse them. A GUI using Electron would be a nice thing, problem is about how to make Electron to work without needing to use X11 or Wayland, just only KMS. Do you have any idea?

**Comment by OrdinaryJellyfish** [comment]
No, not really. I was hoping using X11 or Wayland (preferably X11) would be possible within NodeOS, but I'll try to look up some solutions for this.

**Comment by OrdinaryJellyfish** [comment]
All the results I can find require X11.

**Comment by impetus1** [comment]
There is always OS.js and then there is tightVNC which does in fact leverage x11.

**Comment by piranna** [comment]
Yes, it could be possible to use them and use NodeOS remotely from a browser.
